order-book/src/test/java/com/market/banica/order/book/service/grpc/componentTests/OrderBookComponentIT.java

```diff
@@ -83,31 +96,181 @@
+
+        ItemOrderBookResponse expected = generateItemOrderBookExpectedResponse();
+
+        //Assert
+        assertEquals(expected, auroraResponse);
+    }
+
+    @Test
+    public void calculatorToAuroraToOrderBookRequestsRetryExecutesWithSuccess() {
+        //Arrange
+        List<OrderBookLayer> layers = new ArrayList<>();
+        layers.add(OrderBookLayer.newBuilder().setPrice(productPrice).setQuantity(productQuantity).setOrigin(Origin.AMERICA).build());
+
+        when(itemMarket.getRequestedItem(productName, productQuantity)).thenReturn(layers);
+
+        // Start OrderBook
+        new Thread(() -> {
+            try {
+                Thread.sleep(THREAD_SLEEP_TIME_DEFAULT);
+                grpcCleanup.register(InProcessServerBuilder
+                        .forName(serverName).directExecutor().addService(orderBookService)
+                        .build().start());
+            } catch (InterruptedException | IOException e) {
+                e.printStackTrace();
+            }
+        }).start();
+
+
+        // Start Aurora
+        new Thread(() -> {
+            try {
+                Thread.sleep(THREAD_SLEEP_TIME_DEFAULT);
+                grpcCleanup.register(InProcessServerBuilder
+                        .forName(serverNameThree).directExecutor().addService(new AuroraServiceGrpc.AuroraServiceImplBase() {
+                            @Override
+                            public void request(Aurora.AuroraRequest request, StreamObserver<Aurora.AuroraResponse> responseObserver) {
+                                String[] topicSplit = request.getTopic().split(DELIMITER);
+
+                                ItemOrderBookRequest build = populateItemOrderBookRequest(request.getClientId(), topicSplit[1], Long.parseLong(topicSplit[2]));
+
+                                ItemOrderBookResponse orderBookItemLayers = blockingStub.getOrderBookItemLayers(build);
+
+                                responseObserver.onNext(Aurora.AuroraResponse
+                                        .newBuilder()
+                                        .setMessage(Any.pack(orderBookItemLayers))
+                                        .build());
+                                responseObserver.onCompleted();
+                            }
+                        })
+                        .build().start());
+            } catch (InterruptedException | IOException e) {
+                e.printStackTrace();
+            }
+        }).start();
+
+
+        Aurora.AuroraRequest auroraRequest = Aurora.AuroraRequest.newBuilder().setTopic(orderBookTopicPrefix + productName + DELIMITER + productQuantity).build();
+
+        //Act
+        Aurora.AuroraResponse auroraResponse = auroraBlockingStub.request(auroraRequest);
+
+        Aurora.AuroraResponse expected = Aurora.AuroraResponse.newBuilder().setMessage(Any.pack(generateItemOrderBookExpectedResponse())).build();
```

Comment:
expected should be moved to Arrange part

Reply:
Fixed